### PR TITLE
Review notebook 1 "Pandas is table oriented"

### PR DESCRIPTION
**Pull request by jorisvandenbossche**

### CODE REVIEW

File: notebooks/1_table_oriented.ipynb [hunk 1]
```diff
@@ -0,0 +1,620 @@
+{
```
Comment (TomAugspurger) [inline notebooks/1_table_oriented.ipynb:1]: <p>Do we use column *names* or *labels* more often?</p><p> </p><p>I typically prefer *labels*, since "names" can conflict with Index.name / MultiIndex.names.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='12'/>

Comment (TomAugspurger) [inline notebooks/1_table_oriented.ipynb:1]: <p>Let's provide <code>index=</code> to the DataFrame call above, so that we have an <code>Index</code> object here and not a RangeIndex.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='17'/>

Comment (TomAugspurger) [inline notebooks/1_table_oriented.ipynb:1]: <p>Not quite true about <code>object</code>, especially if https://github.com/pandas-dev/pandas/pull/27949 is merged for 1.0.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='22'/>

Comment (TomAugspurger) [inline notebooks/1_table_oriented.ipynb:1]: <p>typo on <code>Pandas</code> (capital <code>P</code>).</p><p> </p><p>Rather than "single column version", say something like "each column in a DataFrame is a Series".</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='39'/>

Comment (jorisvandenbossche) [reply notebooks/1_table_oriented.ipynb:1]: <p>Even when that is merged, it will not yet be the default, right? </p><p> </p><p>I would reword to somethinglike "any python object, but in practice often means strings"</p>

Comment (jorisvandenbossche) [inline notebooks/1_table_oriented.ipynb:1]: <p>characters -&gt; strings? (I find that a bit more common name, although in SQL land character is maybe often used)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='10'/>

Comment (jorisvandenbossche) [inline notebooks/1_table_oriented.ipynb:1]: <p>Instead of focusing on "a dataframe has attributes", I would maybe rather start with explaining the concepts of column names and index. And then only after that explain that those two "things" are attributes ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='13'/>

Comment (jorisvandenbossche) [inline notebooks/1_table_oriented.ipynb:1]: <p>I suppose the reason that you don't show that a column of a dataframe is a series, is because indexing is only explained in a later tutorial? </p><p>I would maybe consider to just show <code>df[col]</code> already here, which might give a more natural introduction to series = column of dataframe.</p><p> </p><p>And then in a second step we can still show that you can create one manually as well</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='31'/>

**Comment by jorisvandenbossche** [comment]
A general remark on this tutorial is that it is "not very interesting". I am not sure how to make it more, though. 

I fully admit I have this in my tutorials as well, but showing the `.index`, `.columns`, `.dtypes` attributes of a dataframe as a starter is maybe not attractive? I am wondering if we can have some other things in this first notebook, but the difficulty is of course that doing something actually useful directly needs some of the methods that is only seen in later notebooks ..

File: notebooks/1_table_oriented.ipynb (hunk 1 above)
Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <p>ok with using labels. I'll start listing these type of _conventions_ in order to make them consistent in the following tutorials.</p><p> </p>

Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <p>ok, good point</p>

Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <p>As pointed in https://github.com/pandas-dev/pandas/issues/26831#issuecomment-529716754 we will exclude the discussion on object types in the first tutorial.</p>

Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <p>I'll convert all Pandas references with capital P to lowercase in all of the tutorials</p>

Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <p>'each column in a DataFrame is a Series' -&gt; indedd, adjusted</p>

Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <p>i'll make sure adjustments and other tutorials will be adjusted to using <code>strings</code> (cfr. https://github.com/<a href="https://github.com/pandas-dev/pandas/pull/27949" target="_blank">pandas-dev/pandas/pull/27949)</a></p>

Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <p>left discussion on attributes of a tabel out of the starting tutorial to descrease technical elements</p>

Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <p>done</p>

**Comment by stijnvanhoey** [comment]
<p>I refactored with an attempt to keep a balance between:</p><ul><li>introduction to dataframe/series concept</li><li>show that the aim is to DO something with it</li><li>make it more engaing</li><li>keep the technical level low enough</li></ul>

**Comment by stijnvanhoey** [comment]
@datapythonista and @WillAyd can you have a look to see if this improves the first tutorial?

File: notebooks/1_table_oriented.ipynb [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
```
Comment (WillAyd) [inline notebooks/1_table_oriented.ipynb:1]: <p>These images (?) aren't rendering - maybe just an issue with ReviewNB?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='5'/>

Comment (WillAyd) [inline notebooks/1_table_oriented.ipynb:1]: <p>A few comments:</p><p> </p><ul><li>I think would be nice to just use <code>df</code> as the name of this instead of <code>my_dataframe</code></li><li><code>I would defer introducing Categorical types until later</code></li><li><code>We don't need to specify the <code>index=</code> here as it matches the default</code></li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='7'/>

Comment (WillAyd) [inline notebooks/1_table_oriented.ipynb:1]: <p>The Note seems a little strange to me. I get what you are saying but for a new user it doesn't really explain much, so probably best to remove</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='8'/>

Comment (WillAyd) [inline notebooks/1_table_oriented.ipynb:1]: <p>I would delete the second sentence - I think users should be going through sequentially and not hopping around tutorial</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='13'/>

Comment (WillAyd) [inline notebooks/1_table_oriented.ipynb:1]: <p>I'd remove this comment as well - I don't think we care to draw that comparison this early on</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (WillAyd) [inline notebooks/1_table_oriented.ipynb:1]: <p>"Aggregating Data" would be a more appropriate title</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <p>indeed.</p>

Comment (justmarkham) [reply notebooks/1_table_oriented.ipynb:1]: <p>Because <code>df</code> is such a widely used convention for naming DataFrames, I agree with @WillAyd that it would be better to use <code>df</code> than <code>my_dataframe</code>.</p><p> </p><p>I also agree that there's no need to specify the index here. I think that @TomAugspurger suggested specifying the index to avoid a RangeIndex, but the index is no longer being printed out in this notebook, so it's not an issue.</p>

Comment (justmarkham) [reply notebooks/1_table_oriented.ipynb:1]: <p>I think I'm in agreement with @WillAyd's suggestion here. Perhaps don't encourage users to jump ahead to future lessons, though in future lessons it does make sense to link back to previous lessons.</p>

Comment (justmarkham) [reply notebooks/1_table_oriented.ipynb:1]: <p>If you decide to keep this language, I would change "When you are familiar" to "If you are familiar".</p><p> </p><p>Also, "base" should be "based".</p>

Comment (justmarkham) [inline notebooks/1_table_oriented.ipynb:1]: <p>I would use the term "alias" rather than "shortcut".</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='3'/>

Comment (justmarkham) [inline notebooks/1_table_oriented.ipynb:1]: <p>I've only ever seen the word "latter" used when there are two things discussed ("former" and "latter"). Since there are three things being discussed in this case, I would replace "the latter" with "the <code>Pclass</code> column".</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='8'/>

Comment (justmarkham) [inline notebooks/1_table_oriented.ipynb:1]: <p>Typo: "parenthesis" should be "parentheses"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='24'/>

Comment (justmarkham) [inline notebooks/1_table_oriented.ipynb:1]: <p>Rather than saying "methods are like functions", I think we should just say "methods are functions".</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='24'/>

Comment (justmarkham) [inline notebooks/1_table_oriented.ipynb:1]: <p>Typo: "PClass" should be "Pclass"</p><p> </p><p>Also, I think one useful point to mention here is that many pandas operations return a DataFrame or a Series, and this is an example of a pandas operation (describe) returning a Series.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='27'/>

Comment (justmarkham) [inline notebooks/1_table_oriented.ipynb:1]: <p>I've never heard the phrase "besides the looks", so perhaps change the wording?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <ul><li><code>my_dataframe</code> -&gt; <code>df</code>; good point, adjusted</li><li>index is indeed of no more usage, removed</li><li>I added categorical to make sure that when later on applying the <code>describe</code> method it is not taken into account. To overcome this, I now used <code>Sex</code>as a column instead of <code>Pclass</code>. </li></ul>

Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <p>On overcoming too much links towards _future_ tutorials and due to the potential confusion for new users, I removed this note.</p><p> </p>

Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <p>ok, good point!</p>

Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <p>I left the note there for the moment. I do think Python users will _like_ the link with something they already know.</p>

Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <p>I do not want to stress a specific function in that section and use the aggregation just as an example of applying _methods_ in general. As such, I use _do something_ in order to explain the general concept of methods applied on DataFrame/Series.</p>

Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <p>check</p>

Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <p>check.</p>

Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <p>Good point, added to the tutorial.</p>

Comment (stijnvanhoey) [reply notebooks/1_table_oriented.ipynb:1]: <p>Adusted to:</p><p> </p><p>&gt; This is just a starting point. Similar to spreadsheet software, pandas represents data as a table with columns and rows. Apart from the representation, also the data manipulations and calculations you would do in spreadsheet software are supported by pandas. Continue reading the next tutorials to get you started!</p><p> </p><p>Further improvement  welcome</p>

File: notebooks/1_table_oriented.ipynb [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
```
Comment (jorisvandenbossche) [inline notebooks/1_table_oriented.ipynb:1]: <p>But a Series still has a name ... </p><p>Is that something to mention? In the example above we also give it a name.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stijnvanhoey/pandas-getting-started-tutorials/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>
